Adaption to CHERIoT

bacam opened the pull request:
Supported by lowRISC CIC.  @marnovandermaas.

bacam committed:
Update to current Isla

bacam committed:
Update Sail commit

bacam committed:
Very basic CHERIoT support

Produces test files, but no instruction generator and unlikely to handle
the hardware load filter.  Fixed some assumptions about memory sizes.

bacam committed:
Update Isla with -R undefined commit

bacam committed:
Test harness generation for CHERIoT
Currently works with Sail simulator

bacam committed:
Parametrise memory details for CHERIoT

Add cheriot-ibex compatible examples

bacam committed:
Silly cheriot harness typo

bacam committed:
Add option for external instruction generator

bacam committed:
Use helper function for creating cheriot caps that enforces invariant

bacam committed:
Use more general exception stop functions for cheriot

bacam committed:
Update sample cheriot-ibex command

bacam committed:
Tag updates for cheriot
- new `--no-tags-in` option because sonata's load filter bitmap has no tags
- set the alignment mask for tags per-target
- always align tag addresses on memory accesses (e.g., for byte accesses
  that clear tags)

bacam committed:
Apply property to cheriot initial capabilities

Trying to do it in the conversion function doesn't work because we want
to linearise it, which drops assertions.

bacam committed:
Update cheriot test harness
- copy load filter bitmap into place (because the sonata simulator's ELF
  loader won't do it for us)
- correct check for sealing root
- correct setup of wohle address space capabilities
- check whole capability was constructed properly when detecting bad tests
- avoid corner case on initial jump
- sonata uart support

bacam committed:
Add missing sonata example command and configuration

bacam committed:
Add lowRISC notice

marnovandermaas reviewed (commented):
Thanks for putting this PR together. Some of these commits can be squashed, but that is not a huge problem. Would you mind updating the instructions in https://github.com/rems-project/isla-testgen?tab=readme-ov-file#building-and-generating-tests to say how to run the CHERIoT version of isla? I would love to test this code!

bacam committed:
Update README

CHERIoT instructions.  Also update the isla submodule to make them easier,
and expand the sonata-command to make it easier to understand.

bacam commented:
I've added some instructions, including how to produce coverage information.
